src/Visualization/Python/GenerateTetrahedralConnectivity.py

```diff
@@ -0,0 +1,208 @@
+    h5files,
+    subfile_name: str,
+    relative_paths: bool = True,
+    start_time: Optional[float] = None,
+    stop_time: Optional[float] = None,
+    stride: int = 1,
+    coordinates: str = "InertialCoordinates",
+    delete_tetrahedral_connectivity: bool = False,
+):
+    """Generate tetrahedral connectivity using scipy.spatial.Delaunay
+
+    Given the coordinates, this generates a tetrahedral connectivity that can
+    be read in by ParaView or other visualization software (e.g. VisIt or yt).
+    It uses the scipy.spatial.Delaunay class to generate the connectivity, which
+    uses the Qhull library (github.com/qhull/qhull/), which uses the quickhull
+    algorithm and scales as O(n^2) in the worst case. Thus, generating the
+    connectivity can take several minutes per temporal ID. Unfortunately,
+    depending on the particular grid point distribution, generating the
+    connectivity may even fail in qhull, which is unfortunately difficult
+    to fix.
+
+    After the tetrahedral connectivity has been written you can run
+    'generate-xdmf' with the flag '--use-tetrahedral-connectivity'. ParaView
+    volume renderings are sometimes nicer with tetrahedral connectivity and this
+    can be used to fill gaps between finite-difference or Gauss elements.
+
+    If this algorithm is too slow, one possible improvement is to apply
+    qhull to each block of the domain and then connect the blocks to each
+    other separately. This keeps the number of grid points lower for each
+    invocation of qhull, which likely reduces total runtime and may also
+    reduce or eliminate failure cases. In the ideal case we would apply
+    qhull to each element and then connect elements that are using FD or
+    Gauss points to their neighbors.
+
+    \f
+    Arguments:
+      h5files: List of H5 volume data files.
+      subfile_name: Volume data subfile in the H5 files.
+      start_time: Optional. The earliest time at which to start visualizing. The
+        start-time value is included.
+      stop_time: Optional. The time at which to stop visualizing. The stop-time
+        value is not included.
+      stride: Optional. View only every stride'th time step.
+      coordinates: Optional. Name of coordinates dataset. Default:
+        "InertialCoordinates".
+      delete_tetrahedral_connectivity: Optional. Delete the dataset and return.
+        Default: False
+    """
+    h5files = [(h5py.File(filename, "a"), filename) for filename in h5files]
+
+    if not subfile_name:
+        subfiles = available_subfiles(
+            (h5file for h5file, _ in h5files), extension=".vol"
+        )
+        raise RequiredChoiceError(
+            (
+                "Specify '--subfile-name' / '-d' to select a"
+                " subfile containing volume data."
+            ),
+            choices=subfiles,
+        )
+
+    if not subfile_name.endswith(".vol"):
+        subfile_name += ".vol"
+
+    for h5file, filename in h5files:
```

Comment:
You can relatively easily show a progress bar here so the script doesn't appear "frozen" without any indication that it is doing anything. Here's how:

```py
import rich.progress

progress = rich.progress.Progress(
        rich.progress.TextColumn("[progress.description]{task.description}"),
        rich.progress.BarColumn(),
        rich.progress.MofNCompleteColumn(),
        rich.progress.TimeRemainingColumn(),
        disable=(len(h5files) == 1),
    )
task_id = progress.add_task("Processing files", total=len(h5files))

with progress:
    for h5file, filename in progress.track(h5files, task_id=task_id):
        # ...
        progress.update(task_id, advance=1) # don't remember if this is needed
```

Docs are here: https://rich.readthedocs.io/en/stable/progress.html

Reply:
(optional) showing a progress bar over timesteps instead of the log messages would be nicer